src/main/java/uk/gov/hmcts/reform/draftstore/service/DraftService.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.draftstore.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.draftstore.data.DraftStoreDAO;
+import uk.gov.hmcts.reform.draftstore.domain.CreateDraft;
+import uk.gov.hmcts.reform.draftstore.domain.Draft;
+import uk.gov.hmcts.reform.draftstore.domain.DraftList;
+import uk.gov.hmcts.reform.draftstore.domain.UpdateDraft;
+import uk.gov.hmcts.reform.draftstore.exception.AuthorizationException;
+import uk.gov.hmcts.reform.draftstore.exception.NoDraftFoundException;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class DraftService {
+
+    private final DraftStoreDAO draftRepo;
+
+    public DraftService(DraftStoreDAO draftRepo) {
+        this.draftRepo = draftRepo;
+    }
+
+    public Draft read(String id, UserAndService userAndService) {
+        return draftRepo
+            .read(toInternalId(id))
+            .filter(draft -> Objects.equals(draft.userId, userAndService.userId))
+            .filter(draft -> Objects.equals(draft.service, userAndService.service))
+            .orElseThrow(() -> new NoDraftFoundException());
+    }
+
+    public DraftList read(UserAndService userAndService, Integer after, int limit) {
+        List<Draft> drafts =
+            draftRepo.readAll(
+                userAndService.userId,
+                userAndService.service,
+                after,
+                limit
+            );
+        return new DraftList(drafts);
+    }
+
+    public int create(CreateDraft newDraft, UserAndService userAndService) {
+        return draftRepo.insert(userAndService.userId, userAndService.service, newDraft);
+    }
+
+    public void update(String id, UpdateDraft updatedDraft, UserAndService userAndService) {
+        draftRepo
+            .read(toInternalId(id))
+            .map(d -> {
+                assertCanEdit(d, userAndService);
+                draftRepo.update(toInternalId(id), updatedDraft);
+                return d;
+            })
+            .orElseThrow(() -> new NoDraftFoundException());
+    }
+
+    public void delete(String id, UserAndService userAndService) {
+        draftRepo
+            .read(toInternalId(id))
+            .ifPresent(d -> {
+                assertCanEdit(d, userAndService);
+                draftRepo.delete(toInternalId(id));
+            });
+    }
+
+    private void assertCanEdit(Draft draft, UserAndService userAndService) {
```

Comment:
Logic of checking if user and service match the record is again duplicated here. I would suggest to do all of these checks on query level, or have a new component which performs the check.

Reply:
I'll move those two functions (one predicate, one check throwing an exception) to a util class in a future pull request.

Reply:
👍